google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java

```diff
@@ -278,28 +300,46 @@ public void onFacetInitialized(@NotNull Facet facet) {
 
     private final Project project;
     private final String fileName;
-    private final TextFieldWithBrowseButton filePicker;
+    private final TextFieldWithBrowseButton directoryPicker;
     private final Consumer<Path> configFileGenerator;
+    private final boolean isDirectory;
     // Used to refresh the warnings.
     private final Runnable configurationValidator;
 
     GenerateConfigActionListener(
         Project project,
         String fileName,
         Consumer<Path> configFileGenerator,
-        TextFieldWithBrowseButton filePicker,
+        TextFieldWithBrowseButton directoryPicker,
+        boolean isDirectory,
```

Comment:
I'm not really a fan of passing in a boolean like this to switch between the logic.

Is it possible to normalize the directory outside of this class? e.g. if its not a directory, to grab the parent directory outside of here and pass it into this generator?